<launch>
  <arg name="config_file" default="$(find state_filtering)/config/robot_config.yaml"/>

  <param name="use_sim_time" type="bool" value="true" />
  <arg name="camera" default="XTION" />
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find arm_robot_model)/models/DarpaRobotDown.urdf.xacro'" />
  <!-- Pose Estimation Node-->
  <node name="robot_tracker" type="robot_tracker" pkg="state_filtering" output="screen" respawn="false"><!-- launch-prefix="gdb -ex run args">< launch-prefix="valgrind"-->    
    <param name="data_in_meters" type="bool" value="false" />
    <param name="robot_description_package_path" type="string" value="$(find arm_robot_model)" />
    <param name="camera_frame"  type="string"  value="$(arg camera)" /> 
    <rosparam command="load" file="$(arg config_file)" />
    <param name="config_file"  type="string"  value="$(arg config_file)" />   
  </node>

  <!-- This frame has to be published as the camera info in the bag files are relative to XTION_RGB (which should be XTION_IR) -->
  <!-- If this frame is not published, then the CameraDisplay in Rviz won't work-->
  <node pkg="tf" type="static_transform_publisher" name="$(arg camera)_IR_base_link1" args="0 0 0 0 0 0  $(arg camera) /$(arg camera)_RGB 100" />

</launch>
